### PR TITLE
Optimise heading line heights

### PR DESCRIPTION
**Pull request by sarahmonster**
Update line-height for h1-h4 headings as discussed in #7930. Note: I've needed to add specificity to these styles due to a conflict with a line-height being set at 1.8 on `.editor-rich-text__tinymce` elsewhere in the SCSS. I tested and this approach doesn't seem to cause any conflicts, but it's not impossible I missed something!

## Before
screen shot 2018-07-15 at 19 22 34

## After
screen shot 2018-07-15 at 19 29 09

## How has this been tested?
Tested in Firefox/Chrome at different screen sizes.

## Types of changes
Style tweak.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: core-blocks/heading/editor.scss [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 	h2 {
 		font-size: 1.95em;
+		line-height: 1.4;
```
Comment (tofumatt) [inline core-blocks/heading/editor.scss]: It seems weird to specify this here and for the `h2` below as well. Why is this one needed?

**Review by tofumatt — CHANGES_REQUESTED** [review]
I noticed there's an existing line-height variable here:

https://github.com/WordPress/gutenberg/blob/master/edit-post/assets/stylesheets/_variables.scss#L13

and I'm guessing the `line-height: 1.8` you're overriding is declared here: https://github.com/WordPress/gutenberg/blob/master/editor/components/rich-text/style.scss#L91

It might be nice to use an `$editor-line-height-headings` variable to make this easier to change from the variables in the future 🤷‍♂️ 

I'm surprised this selector needs to be so specific.

Visually this looks great, mostly wondering why that extra `h2` declaration is needed and wondering about little tweaks before merging it.

File: core-blocks/heading/editor.scss (hunk 1 above)
Comment (sarahmonster) [reply core-blocks/heading/editor.scss]: That was ... not on purpose.

**Comment by afercia** [comment]
Before touching the line-height values, I'd propose to reconsider the font-size values first. Ideally, the font-size computed values should translate to values in pixels without decimals. Browsers have different font rendering engines and different rounding algorithms. The only way to avoid inconsistencies is to avoid rounding.

I guess the current values have been set with some typographic hierarchy criteria but I don't understand the math behind some values. For example, why H1 is 2.44em? Or H2 1.95em, etc.? Where's the math?

Considering the line-height values depend on the font-size values, adjusting the font-sizes would greatly help. Based on those values, the line-heights can be calculated with the same principle, avoiding computed values with decimals.

Here's how the various values are translated in pixels:
```
h1
2.44em x 16 = 39.04px
39.04 x 1.2 = 54.656px

h2
1.95em x 16 = 31.2px
31.2 x 1.4 = 43.68px

h3
1.56em x 16 = 24.96px
24.96 x 1.4 = 34.944px

h4
1.25em x 16 = 20px
20 x 1.5 = 30px

h5
1em x 16 = 16px
16 x 1.8 = 28.8px

h6
0.8em x 16 = 12.8px
12.8 x 1.8 = 23.04px
```

FWIW, I'd like to see better values also in core 😬

**Comment by tofumatt** [comment]
I agree with your suggestion @afercia, but it's outside the scope of this PR I think 😄

The minor inconsistencies exist before/after the line-height changes here, but if you wanted to move to pixel-based values I am all for that. Could you open an issue for that, including your explanation here? It's super-useful! 👍

**Comment by sarahmonster** [comment]
> It might be nice to use an $editor-line-height-headings variable to make this easier to change from the variables in the future 🤷‍♂️

I'm not opposed to using variables (I am a big fan of them) but I'm not sure how useful they'd be, or if they'd add a great deal of clarity for people. We'd need a minumum of two to account for the different line heights for different headings:

```
$editor-line-height-headings-h1-h3: 1.4;
$editor-line-height-headings-h4: 1.5;
```

If we went that route, it might be better to set variables for each individual heading, so six in total.

> I'm surprised this selector needs to be so specific.

Me also. Unfortunately because the headings don't have their own specific classes, I wasn't able to override the line-height rules for `.editor-rich-text__tinymce.mce-content-body` without adding the `.editor-rich-text__tinymce` to the `h3` selector (or using an `!important` flag, but I'm not a monster.)

Ultimately, it would probably be better to have specific classes for each heading level, and target those classes instead of the actual elements (this helps improve performance), but that'd require a bigger change here. 

I'm open to a better solution if there's something I haven't considered!




I'm surprised this selector needs to be so specific.

**Review by tofumatt — APPROVED** [review]
Alright then, let's :shipit: for now; it's still better than what we had and we can cleanup/improve further later.

**Comment by sarahmonster** [comment]
Oh, and the headings are using a major third scheme for the scale here. Adjusting the font size is definitely out of scope here, but could be worth opening an issue to discuss further. I'd also stick with unitless heights for the line-height, paired with `rem`s for the font size, although there may be a rationale behind using pixels for both instead. 😄

**Comment by tofumatt** [comment]
line-heights [should always be unit-less](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height#Values), but pixels for font-size can be nice/a bit easier to reason about 🤷‍♂️

**Comment by afercia** [comment]
@tofumatt yep I understand it's a bit outside the scope of this PR 🙂Was just considering that if we want to adjust the font-sizes, then we'd need to change also the line-heights set in this PR. Not opposed to open a separate issue.

For clarity, I'm not proposing to use pixel values 😉font-sizes in `em` are fine, but the em values should produce a computed value in pixels without decimals. For example, instead of 
```
h1
2.44em x 16 = 39.04px
```

I'd use
```
h1
2.5em x 16 = 40px
```

I'd also suggest to produce computed pixel values with even numbers, for example `40` instead of `39` and do the same for the line-heights. In my experience, if one of the numbers is even and the other is odd, then browsers may place the font within the line-height in an inconsistent way. Say we have a font of 40 pixels and a line-height of 55 pixels. The remainder is 15. Browsers can't place the font in the middle of the line, they will use ascender and descender spaces of 8 and 7 pixels but there's no guarantee they will do it in the same way. And sub-pixel rendering has still a long way to go.

**Comment by sarahmonster** [comment]
Merging this now since it's better than what we had before—we can discuss computed font sizes in a follow-up issue. 😄

**Pull request merged**